h2/src/test/org/h2/test/db/TestIgnoreCatalogs.java

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright 2004-2019 H2 Group. Multiple-Licensed under the MPL 2.0,
+ * and the EPL 1.0 (https://h2database.com/html/license.html).
+ * Initial Developer: H2 Group
+ */
+package org.h2.test.db;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+import org.h2.api.ErrorCode;
+import org.h2.test.TestBase;
+import org.h2.test.TestDb;
+
+/**
+ * @author aschoerk
+ */
+public class TestIgnoreCatalogs extends TestDb {
+    /**
+     * Run just this test.
+     *
+     * @param a ignored
+     */
+    public static void main(String... a) throws Exception {
+        TestBase.createCaller().init().test();
+    }
+
+    @Override
+    public void test() throws Exception {
+        canCommentOn();
+        canUseDefaultSchema();
+        canYetIdentifyWrongCatalogName();
+        canUseSettingInUrl();
+        canUseSetterSyntax();
+        canCatalogNameEqualSchemaName();
+        canUseCatalogAtIndexName();
+        canCommentOn();
+        canAllCombined();
+        doesNotAcceptEmptySchemaWhenNotMSSQL();
+    }
+
+    public void doesNotAcceptEmptySchemaWhenNotMSSQL() throws SQLException {
+        try (Connection conn = getConnection("ignoreCatalogs;IGNORE_CATALOGS=TRUE;init=drop all objects\\;create schema dbo\\;set schema dbo\\;")) {
```

Comment:
Please also wrap all long lines in this file. Source code can have up to 120 characters per line in H2's sources, comments can have up to 100 characters.

Reply:
Also it's better to remove `INIT` parameters from URLs and execute these commands in normal way. It will improve readability.

Reply:
fixed